### PR TITLE
WIP: Update Avalonia support

### PR DESCRIPTION
**Pull request by danwalmsley**
@vb2ae I am from the Avalonia UI team! This PR targets your exising branch:
1. merges in master latest changed.
2. I have removed the need for ReactiveUI as your sample shouldnt depend on another mvvm framework (maybe I miss understood the need for reactiveui, let me know if this is a mistake)

Im continuing to test so I may add other fixes, and I will also test with latest upstream changes from Avalonia.

### CODE REVIEW

**Comment by Stannieman** [comment]
@danwalmsley FYI you might run into this at some point https://github.com/Caliburn-Micro/Caliburn.Micro/pull/847

Feel free to take it

**Comment by danwalmsley** [comment]
> @danwalmsley FYI you might run into this at some point #847
> 
> Feel free to take it

I'll take a look!

**Comment by danwalmsley** [comment]
> > @danwalmsley FYI you might run into this at some point #847
> > Feel free to take it
> 
> I'll take a look!

@Stannieman do you guys have some chat room or way I can contact you guys? Might smooth communication

**Comment by Stannieman** [comment]
> > > @danwalmsley FYI you might run into this at some point #847
> > > Feel free to take it
> > 
> > 
> > I'll take a look!
> 
> @Stannieman do you guys have some chat room or way I can contact you guys? Might smooth communication

Not that I know of. But I am also just a random guy that helps a hand with Avalonia related stuf from time to time because I happen to use this combo in an app. Maybe the core Caliburn team has some chat.

**Comment by danwalmsley** [comment]
@Stannieman I fixed the issue you pointed out with:  https://github.com/Caliburn-Micro/Caliburn.Micro/pull/846/commits/74bc7f8ba2e9c01c01dad24a3cb7abdfc6e09b9c

I think that is the correct fix, but im also not an expert in Caliburn.

I think that should unblock any other work that is ongoing on avalonia.

**Comment by Stannieman** [comment]
If it is fine that InitializeComponent is not called then yes. But I guess that the default constructor calls it in Avalonia?

**Comment by danwalmsley** [comment]
@Stannieman are you on the avalonia telegram group? I am having trouble getting the normal Feature.Netcore project to build.

https://caliburn-micro.visualstudio.com/caliburn-micro/_build/results?buildId=565&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=b1657ed2-8cf9-53e1-bfff-c661b553db41

**Comment by danwalmsley** [comment]
@Stannieman I posted the wrong build:

https://caliburn-micro.visualstudio.com/caliburn-micro/_build/results?buildId=570&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=4b3c94ee-c80a-5ef5-daaf-a017ad9cae9a

image

**Comment by Stannieman** [comment]
Hmm not sure. It builds for me without touching the code.
You could try changing the URL to http://www.caliburnproject.org which I think is more correct, but again for me it works without that.

**Comment by vb2ae** [comment]
@danwalmsley  I was playing with reactive ui for navigation but removed it on my local.  I will look at the dotnet sample.

File: src/Caliburn.Micro.Avalonia/Caliburn.Micro.Avalonia.csproj [hunk 1]
```diff
@@ -8,20 +8,20 @@
...
-    <PackageReference Include="Avalonia.Desktop" Version="11.0.0-preview5" />
-    <PackageReference Include="Avalonia.Diagnostics" Version="11.0.0-preview5" />
-    <PackageReference Include="Avalonia.Markup.Xaml.Loader" Version="11.0.0-preview5" />
-    <PackageReference Include="Avalonia.ReactiveUI" Version="11.0.0-preview5" />
-    <PackageReference Include="Avalonia.Xaml.Behaviors" Version="11.0.0-preview5" />
-    <PackageReference Include="Avalonia.Xaml.Interactivity" Version="11.0.0-preview5" />
+    <PackageReference Include="Avalonia" Version="$(AvaloniaVersion)" />
+    <PackageReference Include="Avalonia.Desktop" Version="$(AvaloniaVersion)" />
+    <PackageReference Include="Avalonia.Diagnostics" Version="$(AvaloniaVersion)" />
+    <PackageReference Include="Avalonia.Markup.Xaml.Loader" Version="$(AvaloniaVersion)" />
```
Comment (Stannieman) [inline src/Caliburn.Micro.Avalonia/Caliburn.Micro.Avalonia.csproj:22]: Does this still work nicely with the package manager in VS?

**Review by vb2ae — APPROVED** [review]

**Comment by Stannieman** [comment]
> > If it is fine that InitializeComponent is not called then yes. But I guess that the default constructor calls it in Avalonia?
> 
> Yes it does!

@danwalmsley For me this does not work. I either have to call InitializeComponent from constructor in code behind or call it in ViewLocator. Otherwise my views stay blank.

**Pull request merged**